### ISSUE TITLE
Strange behavior of insertion in visual-block mode

### ISSUE THREAD

**Issue by oblitum** [vim/vim#909]
When `virtualedit=all`, characters can't be inserted on blank lines (or past the end of lines) in visual-block mode by using `i`, one needs to use `s`. I don't know the reason of this behaviour, and couldn't find documentation.

Freenode #vim log:

```
23:13 <pepper_chico> strull: just tried here, vim --noplugin. Started empty buffer,
                             :set virtualedit=all
23:13 <pepper_chico> inserted empty lines
23:14 <pepper_chico> then moved to the first line, move to 3rd column
23:14 <pepper_chico> and did the visual block thing
23:14 <pepper_chico> didn't work
23:14 <strull> and "the visual block thing" is ?
23:16 <pepper_chico> strull: enter visual-block editing, select some lines after
                             3rd column, it will highlight just one column, press i
                             to edit the first line, after finishing editing, text is
                             not inserted for the rest
23:16 <pepper_chico> just the first
23:16 <pepper_chico> if you do the same on the 1st column, it works as expected
23:27 <strull> strange. it works sometimes and sometimes it doesn't. Looks like a bug
                             to me
23:28 <strull> aah, try s instead of i
23:29 <strull> I now remembered. That's what I've always done in such situations
23:30 <pepper_chico> will try s
23:30 <pepper_chico> strull: ah, thanks!!!!
23:31 <strull> the behaviour of i and I still look very buggy to me
```

**Comment by chrisbra** [comment]
that is intentionally and explained below `:h v_b_A` (see the Note block).

**Comment by chrisbra** [comment]
oh and please next time please provide clear instructions and do not just paste some random logfile which is hard to follow. Thanks.

**Comment by oblitum** [comment]
@chrisbra many thanks for pointing the correct documentation. Regarding instructions, I've cut and edited what already existed trying to leave just the important parts, removing typos, noise and misinformation, I guess I couldn't redo it from start and end up with better instructions. Just tried my best.

**Comment by oblitum** [comment]
original:

```
23:10 <strull> works for me perfectly
23:13 <pepper_chico> strull: just tried here, vim --noplugin. Started empty buffer, :set virtualedit=all
23:13 <pepper_chico> inserted empty lines
23:14 <pepper_chico> then moved to the first, move to 3rd column
23:14 <pepper_chico> and did the visual block thing
23:14 <pepper_chico> didn't work
23:14 <strull> and "the visual block thing" is ?
23:16 <pepper_chico> strull: enter visual-block editing, select some lines after 3rd column, it will highlight just one column, presse enter to edit the first one, after finishing editing, text is not inserted for the rest
23:16 <pepper_chico> just the first
23:16 <pepper_chico> if you do the same on the 1st column, it works as expected
23:17 -!- Sornaensis is now known as THE_LORD
23:17 <strull> wait ... I can reproduce
23:17 <strull> strange indeed.
23:17 <pepper_chico> =/
23:17 <pepper_chico> that's what I asking for a trick
23:17 <strull> I wonder what setting is the culprit
23:18 <strull> there is one!
23:18 <strull> Because in my setup I have no issue
23:18 <pepper_chico> never looked for it, first time I'm looking for a fix
23:18 <pepper_chico> hmmm
23:18 <pepper_chico> interesting
23:19 <strull> does it work in the gvim for you? And ... at least :set nocp
23:20 <bofh> hi all! how do I process all lines, and remove everything after the '|' character on every line?
23:20 <pepper_chico> doesn't work on gvim
23:21 <pepper_chico> nocp is set too
23:21 <strull> bofh: try %s/|.*
23:21 <strull> bofh: that is  :%s/|.*
23:22 <bofh> strull: I tried this http://stackoverflow.com/questions/569280/vim-search-for-a-pattern-and-if-occurs-delete-to-end-of-line but for some reason it removes every line
23:23 <pepper_chico> strull: if you start an empty vim buffer with your default configuration, you don't have the issue?
23:23 <pepper_chico> but with --noplugin you have?
23:23 <pepper_chico> did you try doing exactly the same with your setup?
23:24 <strull> no
23:24 <strull> from the point you say "press enter to edit" I do instead :
23:24 <strull> press I to start insert mode
23:24 <strull> enter my tex
23:24 <strull> press <esc> to stop
23:25 <strull> err, press i (lower case I) to start insert mode
23:27 <strull> strange. it works sometimes and sometimes it doesn't. Looks like a bug to me
23:28 <strull> aah, try s instead of i
23:29 <strull> I now remembered. That's what I've always done in such situations
23:29 <pepper_chico> strull: no, you didn't figure how it works, the first time you tried you probably didn't setup it properly for not to work. What I've described is exactly the procedure to reproduce the problem. That block-edit doesn't work to insert char when they are going to be inserted past end of line
23:30 <pepper_chico> will try s
23:30 <pepper_chico> strull: ah, thanks!!!!
23:31 <strull> the behaviour of i and I still look very buggy to me
23:31 <pepper_chico> yeah, probably
23:32 <pepper_chico> I agree, it makes no sense
```
